app/controllers/components/course/discussion/topics_component.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+class Course::Discussion::TopicsComponent < SimpleDelegator
+  include Course::ControllerComponentHost::Component
+
+  def self.display_name
+    I18n.t('components.discussion.topics.name')
+  end
+
+  def sidebar_items
+    main_sidebar_items + settings_sidebar_items
+  end
+
+  def settings
+    @settings ||= Course::Discussion::TopicSettings.new(current_course.settings(:discussion_topics))
+  end
+
+  private
+
+  def main_sidebar_items
+    [
+      {
+        key: :discussion_topics,
+        title: settings.title || t('course.discussion.topics.sidebar_title'),
+        weight: 5,
+        path: sidebar_path,
+        unread: unread_count
+      }
+    ]
+  end
+
+  def settings_sidebar_items
+    [
+      {
+        title: t('course.discussion.topics.sidebar_title'),
+        type: :settings,
+        weight: 4,
+        path: course_admin_topics_path(current_course)
+      }
+    ]
+  end
+
+  def sidebar_path
+    if can?(:manage, current_course)
+      pending_course_topics_path(current_course)
+    elsif current_course_user && current_course_user.teaching_assistant?
+      my_students_pending_course_topics_path(current_course)
+    else
+      course_topics_path(current_course)
+    end
+  end
+
+  def unread_count
+    if can?(:manage, current_course)
+      all_unread_count
+    elsif current_course_user && current_course_user.teaching_assistant?
```

Comment:
Sorry, like that -> 

```ruby
if current_course_user && current_course_user.staff? && current_course_user.my_students 
   my_students_unread_count
elsif can?(:manage, current_course)
  all_unread_count
else
  0
end
```

Not sure if this works better, but I thought that if I am a staff and I have `my_students`, I rather the unread count be the pending count (regardless of my course-role: course_owner, course_manager or course_teaching_assistant).

What do you think? Otherwise the initial suggestion works too.